iothub/device/src/ConnectionStatus.cs

```diff
@@ -4,32 +4,36 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// Connection status supported by the client.
+    /// The connection status from a connection change event.
```

Comment:
Reordered this and the other connection status classes to report enums in the general lifecycle order.

Comments all updated too.